### ISSUE TITLE
[AXE-CORE]:  ID attribute value must be unique

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#21788]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix issue.
1. At message the accessibility specialist when issue is resolved.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement
Accessibility specialists discovered Nightwatch axe checks failing silently during a recent staging review. Multiple axe violations were discovered where ID attribute values were not unique (duplicate IDs on a page). Links to pages below.

## Acceptance Criteria
- [ ] All instances of the error are fixed
- [ ] Axe-browser plugin or axe-core CLI scans show 0 Section 508, WCAG2A or WCAG2AA violations

## Environment
* https://www.va.gov/burials-memorials/memorial-items/headstones-markers-medallions/
* https://www.va.gov/burials-memorials/pre-need-eligibility/after-you-apply/
* ~https://www.va.gov/lebanon-health-care/stories/~ #21890

## Steps to Recreate
1. Open one of the listed URLs
1. Open axe browser plugin (assumes already installed in Chrome or Firefox)
1. Confirm the `ID attribute value must be unique` is present one or more times

**Comment by mmiddaugh** [comment]
@1Copenut I pulled the occurrence on https://www.va.gov/lebanon-health-care/stories/ into a separate issue (#21890) since it's part of the VAMC rollout.

I'm not sure who needs to handle the other two pages

**Comment by mmiddaugh** [comment]
Sending to public websites to triage the burials and memorials urls

**Comment by brianalloyd** [comment]
@kelsonic @ncksllvn do you know if this falls under the PW umbrella for support?  If not, who would be the responsible team to action this item.

**Comment by ncksllvn** [comment]
It’s ours to fix I think, since it’s a page under the benefit hubs

**Comment by kelsonic** [comment]
![image](https://user-images.githubusercontent.com/12773166/112489687-4cc1e480-8d44-11eb-98bf-ae87b95ba8b0.png)

![image](https://user-images.githubusercontent.com/12773166/112489741-59463d00-8d44-11eb-8fa5-6f8a1e35313f.png)

I might be missing it but I'm not seeing the axe error `ARIA attributes must conform to valid values` on those pages ^

**Comment by 1Copenut** [comment]
@kelsonic That's my goof. I entered the wrong error into the Acceptance Criteria portion of the ticket. We're looking to eliminate duplicate IDs here.

**Comment by kelsonic** [comment]
Ah perfect! haha no worries at all -- I'll try to get a PR up for this by EOD 👍 Thanks for the clarification!

**Comment by kelsonic** [comment]
These are issues that need to be fixed in Drupal, not `vets-website`. Beth Potts and I are collaborating atm to resolve these duplicate `id` issues, you can follow the thread here if you'd like! 🙂 https://dsva.slack.com/archives/C52CL1PKQ/p1617045145306200

**Comment by ncksllvn** [comment]
Heading IDs are generated in this build step, if they are explicitly added in the CMS editor, https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/modify-dom/add-id-to-subheadings.js. Sounds like it could be a bug with that plugin? this is a weird issue.

On this page, https://www.va.gov/burials-memorials/memorial-items/headstones-markers-medallions/, I see these IDs -

- `enlisted-personnel-who-served-`
- `enlisted-personnel-who-served--100`
- `enlisted-personnel-who-served-` (duplicate)
- `enlisted-personnel-who-served--1`

**Comment by ncksllvn** [comment]
Ohh I see, the CMS content actually does include `enlisted-personnel-who-served-` and `enlisted-personnel-who-served--100`. So the issue is that a couple IDs are hardcoded; a couple are generated. Maybe we can just remove the two from the CMS content.

**Comment by bethpottsVADEPO** [comment]
@kelsonic @1Copenut @ncksllvn @brianalloyd I fixed the headstones page and Selina fixed the after you apply for an eligibility determination page. The headstones one should be live today--I think that one made it into deployment. The other one will be live tomorrow.

**Comment by bethpottsVADEPO** [comment]
@lunascoop Can you put a note here once the "After you apply for an eligibility determination" edits you made yesterday are live? That will close the loop on this work. 

I see that the other edits to the headstones page are live already. So we just have to validate the other page, then can close this ticket.

**Comment by lunascoop** [comment]
The updates to the After you apply for an eligibility determination are live. https://www.va.gov/burials-memorials/pre-need-eligibility/after-you-apply/

**Comment by bethpottsVADEPO** [comment]
@1Copenut I think we're good to close this ticket, but wanted to check with you to make sure you don't also need to validate before we do so!

**Comment by 1Copenut** [comment]
Agree @bethpotts. Violations are gone on the two relevant pages and the new issue ticket has been added to my epic. I'll move this one to close.
